front/src/components/User/UsersTable.tsx

```diff
@@ -0,0 +1,204 @@
+    }
+    const head = array.slice(0, size);
+    const tail = array.slice(size);
+    return [head, ...chunk(tail, size)];
+  }
+
+  const pages = chunk(
+    Array(sortedUsers.length)
+      .fill(0)
+      .map((_, index) => ({
+        ...sortedUsers[index],
+      })),
+    perPage,
+  );
+
+  const rows = pages[page - 1].map((user) => (
+    <TableUserRow
+      previousUrl={previousUrl}
+      key={user.uid}
+      user={user}
+      submitDeletion={submitDeletion}
+    />
+  ));
+
+  return <>
+    <TableWithFilter
+      headers={[
+        'uid',
+        'email',
+        'firstname',
+        'lastname',
+        'createdAt',
+        'updatedAt',
+        'deletedAt',
+        'isAdmin',
+        'actions',
+      ]}
+      onHeaderClick={handleSort}
+      sortColumn={sortColumn}
+      sortOrder={sortOrder}
+      sortSettings={sortSettings}
+      rows={rows}
+    />
+    <PaginationMantine
+      total={pages.length} value={page} onChange={setPage}
+    />
+  </>;
+}
+
+function TableUserRow({
+  user,
+  previousUrl,
+  submitDeletion,
+}: WithPreviousUrl<{
+  user: User
+  submitDeletion: (uid: string) => Promise<void>
+}>) {
+  return <Table.Tr>
+    <Table.Td>{user.uid}</Table.Td>
+    <Table.Td>{user.email}</Table.Td>
+    <Table.Td>{user.firstname}</Table.Td>
+    <Table.Td>{user.lastname}</Table.Td>
+    <Table.Td>{new Date(user.createdAt).toDateString()}</Table.Td>
+    <Table.Td>{user.updatedAt ? new Date(user.updatedAt).toDateString() : ''}</Table.Td>
+    <Table.Td>{user.deletedAt ? new Date(user.deletedAt).toDateString() : ''}</Table.Td>
+    <Table.Td>{user.isAdmin ? 'admin' : 'user'}</Table.Td>
+    <Table.Td className="text-right">
+      <TableItemMenu
+        user={user}
+        previousUrl={previousUrl}
+        submitDeletion={submitDeletion}
+      />
+    </Table.Td>
+  </Table.Tr>;
+}
+
+function TableItemMenu({
+  user,
+  previousUrl,
+  submitDeletion,
+}: WithPreviousUrl<{
+  user: User
+  submitDeletion: (uid: string) => Promise<void>
+}>) {
+  const [openedAction, setOpenedAction] = useState(false);
+  const [openedDeleteModale, { open, close }] = useDisclosure(false);
+
+  return (
+    <>
+      <Menu shadow="md" width={200} opened={openedAction} onChange={setOpenedAction}>
+        <Menu.Target>
+          <Button
+            component="div"
+            variant="link"
+            size="xs"
+            color="blue"
+          >
+            {trans('pages.admin.user.list.table.action.button')}
+          </Button>
+        </Menu.Target>
+        <Menu.Dropdown>
+          <Menu.Item>
+            <Link to={route(UserUpdate, { id: user.uid })} state={{ previousUrl }}>
+              {trans('pages.admin.user.list.table.action.update')}
+            </Link>
+          </Menu.Item>
+          <Menu.Item>
+            <a
+              onClick={() => {
+                open();
+              }}
+            >{trans('pages.admin.user.list.table.action.delete')}</a>
+          </Menu.Item>
```

Comment:
No need to put a child component like that, Mantine have polymorphic components for this use case (Make the styling better):
```suggestion
          <Menu.Item
            component={Link}
            to={route(UserUpdate, { id: user.uid })}
            state={{ previousUrl }}
          >
            trans('pages.admin.user.list.table.action.update')}
          </Menu.Item>
          <Menu.Item
            onClick={() => {
              open();
            }}
          >
            {trans('pages.admin.user.list.table.action.delete')}
          </Menu.Item>
```